plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -36,8 +38,29 @@ type Query struct {
 // MapQuery type
 type MapQuery map[string]Query
 
+// HealthMetric struct tracking the number of attempted vs successful connections for each connection string
+type HealthMetric struct {
+	AttemptedQueries  int
+	SuccessfulQueries int
+}
+
 const defaultServer = "Server=.;app name=telegraf;log=1;"
 
+const (
+	typeAzureSQLDB              = "AzureSQLDB"
+	typeAzureSQLManagedInstance = "AzureSQLManagedInstance"
+	typeSQLServer               = "SQLServer"
+)
+
+const (
+	healthMetricName              = "sqlserver_telegraf_health"
+	healthMetricInstanceTag       = "connection_sql_instance"
+	healthMetricDatabaseTag       = "connection_database_name"
+	healthMetricAttemptedQueries  = "AttemptedQueries"
+	healthMetricSuccessfulQueries = "SuccessfulQueries"
+	healthMetricDatabaseType      = "DatabaseType"
+)
```

Comment:
I'm still not sure I like this definitions. You use it in one place in the code, so it seems to me you are adding too much lines here, on the other hand you have everything in one place... As it is a matter of taste, please proceed as you prefer...

However, please unify the label styles here. I don't think having snake-case (tags) and CamelCase (fields) in one metric is very intuitive.

Reply:
Good gatch on the snake-case vs. CamelCase. I changed that. 

I added these as constants here because I wanted to reference them in the sqlserver_test.go. It made testing a bit cleaner.

Reply:
Fair enough. Please close as resolved.